### PR TITLE
Docs: Organize and update block theme docs

### PR DESCRIPTION
**Pull request by mkaz**
## Description

A follow up PR to #31065

Updates and organizes the block theme documentation. An overview document was missing from the index, plus naming and consistency around using "block theme"

The previous themes/ directory was removed it contained an extra layer to click through to get to the single theme-support file, plus having two sections about themes is confusing.

- Move unlinked overview document to `how-to-guides/themes/block-theme-overview.md`
- Move previous `block-theme/README.md` to `themes/create-block-theme.md`
- Update titles to simplify in sidebar and reflect new names
- Update references of "block-based theme" to "block theme"
- Search through repo and update incoming links

## How has this been tested?

Documentation update.

- Confirm new organization makes sense
- Confirm titles and minor text changes are accurate
- Confirm links work

You can browse the changes [on the branch rendered here](https://github.com/WordPress/gutenberg/tree/docs/block-theme-dir/docs/how-to-guides).

## Types of changes

- Documentation
- Updates toc.json and manifest.json
- Meta Trac Ticket for redirects at: https://meta.trac.wordpress.org/ticket/5714

### CODE REVIEW

**Comment by carolinan** [comment]
theme-support.md feels a bit left out. If `how-to-guides/themes/` was used, then that would work for all types of themes and it could eventually also house information about converting and universal/hybrid themes.

**Comment by mkaz** [comment]
@carolinan I think you're right, I didn't like that there were two Theme sections and was focused on the Block Theme work. Stepping back it makes sense to make one more encompassing directory and move the block theme work there. It also simplifies changes, I'll update the PR to reflect.

**Comment by mkaz** [comment]
I updated the PR and description to move files together under a unified `themes/` directory.
I also updated the corresponding Trac ticket with redirects and various links. 👍

**Review by JustinyAhin — APPROVED** [review]
The changes themselves look fine to me. I've searched across the branch's code if all the links have been replaced, and they are fine as well.

@mkaz I guess you already coordinated with someone from Meta to commit your patch

File: docs/how-to-guides/themes/theme-json.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
-# Themes & Block Editor: experimental theme.json
+# Global Settings (theme.json)
```
Comment (oandregal) [inline docs/how-to-guides/themes/theme-json.md:1]: I still struggle to communicate `theme.json` is both for settings & styles. Perhaps naming this "Global Settings & Styles" would help in communicating that?

To clarify: given this doesn't have any effects in URLs, etc. and can be reverted/updated/etc, I'm fine experimenting with anything (this is not a blocker for me).

Comment (mkaz) [reply docs/how-to-guides/themes/theme-json.md:1]: I read it as styles is one of the types of Global settings you can set, but not sure if that is too nuanced.

I'm open to suggestions, but do think we should name the file a single thing: "Global Properties" ? 🤷‍♂️

**Pull request merged**